src/Features/CSharp/Portable/Snippets/CSharpConsoleSnippetProvider.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Snippets
+{
+    [ExportSnippetProvider(nameof(ISnippetProvider), LanguageNames.CSharp), Shared]
+    internal class CSharpConsoleSnippetProvider : AbstractConsoleSnippetProvider
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public CSharpConsoleSnippetProvider()
+        {
+        }
+
+        protected override bool ShouldDisplaySnippet(SyntaxContext context)
+        {
+            var csharpContext = (CSharpSyntaxContext)context;
+            var token = context.LeftToken;
+            var isInNamespace = token.GetAncestors<SyntaxNode>()
+                .Any(node => node.IsKind(SyntaxKind.NamespaceDeclaration) ||
+                             node.IsKind(SyntaxKind.FileScopedNamespaceDeclaration));
+
+            return csharpContext.IsStatementContext || (csharpContext.IsGlobalStatementContext && !isInNamespace);
+        }
+
+        protected override SyntaxNode? GetAsyncSupportingDeclaration(SyntaxToken token)
+        {
+            var node = token.GetAncestor(node => node.IsAsyncSupportingFunctionSyntax());
+            if (node is LocalFunctionStatementSyntax { ExpressionBody: null, Body: null })
+            {
+                return node.Parent?.FirstAncestorOrSelf<SyntaxNode>(node => node.IsAsyncSupportingFunctionSyntax());
+            }
+
+            return node;
+        }
+
+        private static SyntaxNode? GetConsoleExpressionStatement(SyntaxNode root, int position)
+        {
+            var closestNode = root.FindNode(TextSpan.FromBounds(position, position));
+            return closestNode.GetAncestorOrThis<ExpressionStatementSyntax>();
+        }
+
+        protected override Task<ImmutableArray<TextSpan>> GetRenameLocationsAsync(Document document, int position, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(ImmutableArray<TextSpan>.Empty);
+        }
```

Comment:
consider makign this the default behavior of the base type so that snippets only have to do this if htey really want to rename.